javascript/reflexes.js

```diff
@@ -0,0 +1,207 @@
+import CableReady from 'cable_ready'
+import Debug from './debug'
+import { dispatchLifecycleEvent } from './lifecycle'
+import { XPathToElement, debounce, emitEvent } from './utils'
+import { allReflexControllers, findControllerByReflexName } from './controllers'
+import { attributeValue, attributeValues } from './attributes'
+import isolationMode from './isolation_mode'
+
+const reflexes = {}
+
+export default reflexes
+
+export const performOperations = data => {

```

Comment:
It would also be very nice if this could be exported from `stimulus_reflex.js`.

Reply:
That's a good idea, although as someone just learning this stuff... wouldn't it make more sense to just import them from `reflexes.js`?

BTW - are you talking about `reflexes`, `performOperations` or both?

Reply:
Ah, yes. Exporting the entire module `reflexes.js` from stimulus_reflex.js might make more sense. From my point of view someone just learning this stuff shouldn't really need to learn about this since this is still pretty much internals of stimulus_reflex. Exposing these would help in being able to build on top of stimulus_reflex in a better way.

Reply:
Exporting `reflexes` actually makes good sense because it is the new location for what used to be `window.reflexes` - it's where all of the magic happens.

Exporting all of the functions in reflex.js through the main file to save someone implementing plugins to SR from having to import both is not a strategy that makes sense to me. If you need the functions from a file in a library, you import it. This is not an unreasonable expectation of someone building a transport plugin. You need to be at least comfortable with the internals, that's not something we can hide away.